extensions/shared/external-media/media-browser/index.js

```diff
@@ -63,14 +63,22 @@ function MediaBrowser( props ) {
 		[ className ]: true,
 	} );
 
+	const prevMediaCount = useRef();
+
+	const onLoadMoreClick = () => {
+		prevMediaCount.current = media.length;
+		nextPage();
+	};
+
 	return (
 		<div className={ wrapper }>
 			<ul className={ classes }>
-				{ media.map( item => (
+				{ media.map( ( item, index ) => (
 					<MediaItem
 						item={ item }
 						key={ item.ID }
 						onClick={ onSelectImage }
+						focusOnMount={ index === prevMediaCount.current }
```

Comment:
What do you think of this method? Is this too over the top? I'm not sure if this set-up could eventually have an edge case or not. 

https://github.com/Automattic/jetpack/pull/15894/commits/ba70e7af7d4fe89003293114154f40a75f12d272

Reply:
I think this implementation already covers mentioned edge-cases. In https://github.com/Automattic/jetpack/commit/ba70e7af7d4fe89003293114154f40a75f12d272 you wrote:

> We only want to send focus to an image if:
>  1. previousMediaCount is not 0 (we don't want to focus the first image on initial modal load)
>  2. new images have been added to to the media (we don't want to focus the previous count if no new images were found)

**Case `#1`:** Don't focus on the initial load.

The initial `prevMediaCount.current` is `undefined` until the `Load More` is clicked, meaning nothing will be focused initially because the value is not a number ✅

**Case `#2`:** Don't focus if no additional media items are loaded.

Let's say we have `10` images and the `prevMediaCount.current` is `5`. The `Load More` is clicked again and the `prevMediaCount.current` is set to `10`. No additional items are loaded - the media count is still `10`. The item with index `10` doesn't exist (`9` is the highest), so nothing gets focused ✅

---

Let me know if that makes sense. I hope I didn't miss anything there 🤞

Reply:
Made the conditional a bit more specific in https://github.com/Automattic/jetpack/pull/15868/commits/562652b20aa7d9db1b3370b1e713b19ecdca01bd.